ptf/sairoute.py

```diff
@@ -0,0 +1,1118 @@
+            ip_ttl=64)
+        exp_pkt = simple_tcp_packet(
+            eth_dst='00:11:22:33:44:55',
+            eth_src=ROUTER_MAC,
+            ip_dst='10.10.10.1',
+            ip_src='192.168.0.1',
+            ip_id=105,
+            ip_ttl=63)
+
+        try:
+            print("Sending packet port %d to ports %d, %d, %d" %
+                  (self.dev_port10, self.dev_port24, self.dev_port25,
+                   self.dev_port26) + "(192.168.0.1 -> 10.10.10.1) Routed")
+            send_packet(self, self.dev_port10, pkt)
+            verify_packets(self, exp_pkt, [self.dev_port24, self.dev_port25,
+                                           self.dev_port26])
+
+            mac_action = SAI_PACKET_ACTION_FORWARD
+            fdb_entry1 = sai_thrift_fdb_entry_t(
+                switch_id=self.switch_id, mac_address=dmac1, bv_id=vlan100)
+            sai_thrift_create_fdb_entry(
+                self.client, fdb_entry1, type=SAI_FDB_ENTRY_TYPE_STATIC,
+                bridge_port_id=port24_bp, packet_action=mac_action)
+            fdb_entry2 = sai_thrift_fdb_entry_t(
+                switch_id=self.switch_id, mac_address=dmac2, bv_id=vlan100)
+            sai_thrift_create_fdb_entry(
+                self.client, fdb_entry2, type=SAI_FDB_ENTRY_TYPE_STATIC,
+                bridge_port_id=port25_bp, packet_action=mac_action)
+            fdb_entry3 = sai_thrift_fdb_entry_t(
+                switch_id=self.switch_id, mac_address=dmac3, bv_id=vlan100)
+            sai_thrift_create_fdb_entry(
+                self.client, fdb_entry3, type=SAI_FDB_ENTRY_TYPE_STATIC,
+                bridge_port_id=port26_bp, packet_action=mac_action)
+
+            print("Sending packet port %d to port %d " %
+                  (self.dev_port10, self.dev_port24) +
+                  "(192.168.0.1 -> 10.10.10.1) Routed")
+            send_packet(self, self.dev_port10, pkt)
+            verify_packets(self, exp_pkt, [self.dev_port24])
+
+            sai_thrift_remove_fdb_entry(self.client, fdb_entry3)
+            sai_thrift_remove_fdb_entry(self.client, fdb_entry2)
+            sai_thrift_remove_fdb_entry(self.client, fdb_entry1)
+
+        finally:
+            sai_thrift_remove_route_entry(self.client, route_entry3)
+            sai_thrift_remove_route_entry(self.client, route_entry2)
+            sai_thrift_remove_route_entry(self.client, route_entry1)
+
+            sai_thrift_remove_neighbor_entry(self.client, neighbor_entry3)
+            sai_thrift_remove_neighbor_entry(self.client, neighbor_entry2)
+            sai_thrift_remove_neighbor_entry(self.client, neighbor_entry1)
+
+            sai_thrift_remove_next_hop(self.client, nhop3)
+            sai_thrift_remove_next_hop(self.client, nhop2)
+            sai_thrift_remove_next_hop(self.client, nhop1)
+
+            sai_thrift_remove_router_interface(self.client, vlan100_rif)
+            sai_thrift_set_port_attribute(self.client, self.port24,
+                                          port_vlan_id=1)
```

Comment:
Please replace all instances in tearDown where we reset port_vlan_id=1 to port_vlan_id=0